es_base_block.py

```diff
@@ -1,12 +1,64 @@
+import ast
+from enum import Enum
+import logging
+
 from nio.common.block.base import Block
 from nio.common.command.params.dict import DictParameter
 from nio.common.command.params.string import StringParameter
+from nio.metadata.properties import ListProperty, SelectProperty, \
+    PropertyHolder, IntProperty
```

Comment:
Let's include the String and Expression properties in this list too, for consistency